backend/api/Areas/Keycloak/Controllers/UserController.cs

```diff
@@ -0,0 +1,107 @@
+using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Pims.Api.Helpers;
+using Pims.Api.Helpers.Extensions;
+using Pims.Dal.Exceptions;
+using Pims.Dal.Services.Admin;
+using Model = Pims.Api.Models.Keycloak;
+
+namespace Pims.Api.Areas.Keycloak.Controllers
+{
+    /// <summary>
+    /// UserController class, provides endpoints for managing users within keycloak.
+    /// </summary>
+    [Authorize(Roles = "system-administrator")]
+    [ApiController]
+    [Area("keycloak")]
+    [Route("/api/[area]/users")]
+    public class UserController : ControllerBase
+    {
+        #region Variables
+        private readonly ILogger<UserController> _logger;
+        private readonly IPimsAdminService _pimsAdminService;
+        private readonly IMapper _mapper;
+        private readonly IConfiguration _configuration;
+        private readonly IKeycloakRequestClient _requestClient;
+        #endregion
+
+        #region Constructors
+        /// <summary>
+        /// /// Creates a new instance of a UserController class.
+        /// </summary>
+        /// <param name="logger"></param>
+        /// <param name="pimsAdminService"></param>
+        /// <param name="mapper"></param>
+        /// <param name="configuration"></param>
+        public UserController(ILogger<UserController> logger, IPimsAdminService pimsAdminService, IMapper mapper, IConfiguration configuration, IKeycloakRequestClient requestClient)
+        {
+            _logger = logger;
+            _pimsAdminService = pimsAdminService;
+            _mapper = mapper;
+            _configuration = configuration;
+            _requestClient = requestClient;
+        }
+        #endregion
+
+        #region Endpoints
+        /// <summary>
+        /// Fetch a list of users from keycloak.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        public async Task<IActionResult> GetUsersAsync()
+        {
+            var authority = _configuration.GetValue<string>("Keycloak:Admin:Authority") ??
+                throw new ConfigurationException("The configuration for Keycloak:Admin:Authority is invalid or missing.");
+            var users_url = _configuration.GetValue<string>("Keycloak:Admin:Users") ??
+                throw new ConfigurationException("The configuration for Keycloak:Admin:Users is invalid or missing.");
+            var response = await _requestClient.GetAsync($"{authority}{users_url}");
+
+            return await response.HandleResponseAsync<IEnumerable<Model.UserModel>>();
+        }
+
+        /// <summary>
+        /// Fetch the user for the specified 'id'.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetUserAsync(string id)
+        {
+            var authority = _configuration.GetValue<string>("Keycloak:Admin:Authority") ??
```

Comment:
Could these be moved to the constructor? or perhaps a common method.

Reply:
Agree with Devin